test/PositionManager.ts

```diff
@@ -131,10 +191,201 @@ describe("GammaPoolFactory", function () {
                 expect(event.args.pool).to.equal(gammaPool.address);
                 expect(event.args.assets.toNumber()).to.equal(16);
             });
-        });/**/
+        });
+
+        it("depositReserves", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            const DepositReservesParams =  {
+                cfmm: cfmm.address,
+                amountsDesired: [10000, 100],
+                amountsMin: [1000, 10],
+                to: addr4.address,
+                protocol: 1,
+                deadline: ethers.constants.MaxUint256
+            }
+            const res = await (await posMgr.depositReserves(DepositReservesParams)).wait();
+            res.events.forEach(function(event: any, i: any){
+                expect(event.args).to.not.be.an('undefined');
+                console.log("event >>");
+                console.log(event.args);
+                expect(event.args.pool).to.equal(gammaPool.address);
+                expect(event.args.reservesLen).to.equal(4);
+                expect(event.args.shares.toNumber()).to.equal(18);
+            });
+        });
+
+        it("withdrawReserves", async function () {            
+            await gammaPool.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
```

Comment:
Should make this approval in the beforeEach function

Reply:
In factory contract?

Reply:
yes probably there